tests/primeqa/boolqa/processors/postprocessors/test_extractive_pipeline_postprocessor.py

```diff
@@ -0,0 +1,156 @@
+
+                end_scores_iter = iter(mock_logits_expected_predictions['end_scores'][example_idx][feat_idx])
+                for i, offset in enumerate(offset_mapping):
+                    if i == 0:
+                        start_logits.append(mock_logits_expected_predictions['start_score_of_cls_token'])
+                        end_logits.append(mock_logits_expected_predictions['end_score_of_cls_token'])
+                    elif offset == None:
+                        start_logits.append(mock_logits_expected_predictions['score_none_token'])
+                        end_logits.append(mock_logits_expected_predictions['score_none_token'])
+                    else:
+                        start_logits.append(next(start_scores_iter))
+                        end_logits.append(next(end_scores_iter))
+
+                start_logits +=  [sys.float_info.min]*(128-len(start_logits))
+                all_start_logits.append(np.array( start_logits))
+                end_logits += [sys.float_info.min]*(128-len(end_logits))
+                all_end_logits.append(np.array( end_logits))
+                all_target_type_logits.append(mock_logits_expected_predictions['target_type_scores'][example_idx])
+
+        return adjusted_start_end_index, ( np.array(all_start_logits), np.array(all_end_logits), np.array(all_target_type_logits) )
+
+    def test_post_processor_has_examples_and_features(self, eval_examples_and_features, mock_logits_expected_predictions):
+        eval_examples, eval_features = eval_examples_and_features
+        postprocessor_class = ExtractivePipelinePostProcessor  
+        scorer_type='weighted_sum_target_type_and_score_diff'
+
+        expected_start_end_index, predictions = self._start_end_target_type_logits(eval_examples, eval_features, mock_logits_expected_predictions)
+        
+        postprocessor = postprocessor_class(k=5, n_best_size=3, max_answer_length=30,
+                                            scorer_type=SupportedSpanScorers(scorer_type),
+                                            single_context_multiple_passages=False)
+        example_predictions = postprocessor.process(eval_examples, eval_features, predictions)
+        print('example_predictions='+str(example_predictions))
+        for example_id, preds in  example_predictions.items():
+            predicted = preds[0]
+            expected = mock_logits_expected_predictions['expected_predictions'][example_id]
+            expected_start_end = expected_start_end_index[example_id]
+            ptargettype = int(np.argmax(predicted['target_type_logits']))
+            assert (predicted['question'] == expected['question'] )
+            assert (predicted['language'] == expected['language'] )
+            assert (predicted['start_index'], predicted['end_index']) == expected_start_end
+            assert predicted['passage_index']  == expected['passage_index'] 
+            assert ptargettype  == expected['target_type'] 
+
```

Comment:
Maybe take a look at the SQUAD postprocessor test where it only checks the additional fields
https://github.com/primeqa/primeqa/blob/c6cf549bcc2e0e4fde1cfbfa31aa8a75c73c06be/tests/primeqa/mrc/processors/postprocessors/test_squad_postprocessor.py#L94

Reply:
it seems safer to me to have a few extra asserts in the unit tests

Reply:
it seems safer to me to have a few extra asserts for the unit tests